src/qp/metrics/metrics.py

```diff
@@ -274,16 +282,20 @@ def calculate_anderson_darling(p, scipy_distribution='norm', num_samples=100, _r
         logging.warning("Each element in the ensemble `p` must be a single distribution.")
 
    # Pass an array of random variables and the name of a scipy distribution to the quick anderson darling function
+    p_rvs = p.rvs(size=num_samples, random_state=_random_state)
```

Comment:
The changes here, and on lines 337 and 391 are performance improvements. While each of these functions are marked as `Deprecated`, we want to continue to use them, at least in the short term as a way to compare the Scipy 1.9 metrics to Scipy 1.10 metrics. 

For the size of the tests that we're running, these changes don't result is a large difference in runtime. But anyone who wants to do larger scale testing, will likely appreciate these modifications.